### PR TITLE
Fix/TR-1098 implement review mode file upload interaction

### PR DESCRIPTION
**Pull request by marpesia**
Related to [https://oat-sa.atlassian.net/browse/TR-1098](https://oat-sa.atlassian.net/browse/TR-1098)

- Remove the option of upload a file on Review mode
- Add option of download the uploaded file
- Add a preview of the uploaded file

Test:
- Upload a file on a file upload item test
- Complete the test and go to Results tab and check on the Review view
- You should can download the file and see a preview of it

### CODE REVIEW

**Review by jsconan — CHANGES_REQUESTED** [review]
Nice job!
It is working fine:
image

However, I have a few remarks on the implementation, please see my comments below.
Please also make sure all your commits are following the conventions. Unfortunately, the last one doesn't. You can fix it by rewriting the history (but it will require a force push).

**Comment by marpesia** [comment]
All fixed.
There are 3 views that get data from UploadInteractions.js:
- Create item (qtiItem)
- Test-taker (qtiCommonRenderer)
- Review (reviewRenderer)

The first view calls to resetGui function so I must keep without changes.
The translations has set on templates as it is no longer calling to resetGui function on the Test-taker and the Review side because the override function.

File: src/reviewRenderer/renderers/interactions/UploadInteraction.js [hunk 1]
```diff
@@ -22,5 +22,78 @@
...
+            .empty()
+            .text(filename);
+
+        $container
+            .find('.btn-download')
+            .click(function (e) {
+                e.preventDefault();
+                downloadLink.href = downloadUrl;
+                downloadLink.download = filename;
+                downloadLink.click();
```
[the remark this answers is not in the document]
Comment (marpesia) [reply src/reviewRenderer/renderers/interactions/UploadInteraction.js:60]: This is the only way I have found.

File: src/reviewRenderer/renderers/interactions/UploadInteraction.js [hunk 2]
```diff
@@ -22,5 +22,75 @@
...
+        $previewArea.css({
+            'min-width': '300px',
+            'min-height': '200px'
+        })
+    }
+    interaction.data('_response', response);
+};
+
+function render(interaction) {
+    callResetGui(interaction);
```
Comment (jsconan) [inline src/reviewRenderer/renderers/interactions/UploadInteraction.js:82]: **thought:** I wonder if the following should not be needed as well:
```javascript
    //init response
    interaction.data('_response', { base: null });
```

File: src/reviewRenderer/renderers/interactions/UploadInteraction.js [hunk 3]
```diff
@@ -22,5 +22,75 @@
...
+    }
+}
 
-export default Object.assign({}, uploadInteraction, {template});
\ No newline at end of file
+export default Object.assign({}, uploadInteraction, {
+    template,
+    setResponse,
+    render
+});
```
Comment (jsconan) [inline src/reviewRenderer/renderers/interactions/UploadInteraction.js:96]: **thought:** I wonder if the function `resetGui` should not be added as well? Couldn't the existing one conflict with the preview mode (`Browse` text apply on the `.btn-info` element).

Comment (marpesia) [reply src/reviewRenderer/renderers/interactions/UploadInteraction.js:96]: I've check all views and everything goes well. I think it is no needed

**Comment by marpesia** [comment]
Change approved by UX

**Review by btamas — APPROVED** [review]
**Review Checklist**
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful
- [ ] Pull request's target is not `master`
- [ ] Changelog is updated according to changes (if applicable)
- [ ] Documentation is updated according to changes (if applicable)

**Pull request merged**